packages/devtools-reps/src/object-inspector/utils/client.js

```diff
@@ -80,6 +80,19 @@ async function getPrototype(
   return objectClient.getPrototype();
 }
 
+async function getFullText(
+  objectClient: ObjectClient,
+  stringGrip: Node,
+) : ?Promise<{prototype?: Object}> {
+  if (typeof objectClient.getString !== "function") {
```

Comment:
It doesn't seem like we have the `getString` function on the [ObjectClient](https://searchfox.org/mozilla-central/source/devtools/shared/client/object-client.js)
But we do have it on the [WebConsoleClient](https://searchfox.org/mozilla-central/rev/028cd8342899357d80fafba2d528a0fd5819e316/devtools/shared/webconsole/client.js#709-745).
Another way of getting a longString content is to use the [LongStringClient](https://searchfox.org/mozilla-central/source/devtools/shared/client/long-string-client.js)

So here, either we can pass to the ObjectInspector a `createLongStringClient` function, like we do for `createObjectClient`, which would create the LongStringClient and call substring on it, or a more straightforward  `getLongStringFullText` function (which could use `WebConsoleClient.getString` in caller code).

What do you think would be the best ?

Reply:
I think I'll go with `WebConsoleClient.getString` since it seems to be kind of optimized and avoids redundant queries by immediately returning the cached `_fullText`. It is also more straightforward to use since it gets it through 1 public method call.